pandas/io/formats/csvs.py

```diff
@@ -188,6 +198,43 @@ def save(self):
                 for _fh in handles:
                     _fh.close()
 
+    def _index_label_encoder(self):
+        """
+        Encode index label if it is not False.
+
+        Returns
+        -------
+        index_label: list
+            New index_label given index types
+        encode_labels: list
+            List of index labels
+        """
+        index_label = self.index_label
+        obj = self.obj
+
+        if index_label is True:
+            # append index label based on index type
+            if isinstance(obj.index, ABCMultiIndex):
+                index_label = []
```

Comment:
There's generally a lot of shadowing of `index_label` here that I think can only cause confusion. Given this always needs to be a list on return, why not just insatiate to an empty list and then append the appropriate value within each branch?

Reply:
i simplified a bit, would like to hear your opinion. @WillAyd